src/Microsoft.TemplateEngine.Core/Include.cs

```diff
@@ -8,11 +8,12 @@ namespace Microsoft.TemplateEngine.Core
 {
     public class Include : IOperationProvider
     {
-        public Include(string startToken, string endToken, Func<string, Stream> sourceStreamOpener)
+        public Include(string startToken, string endToken, Func<string, Stream> sourceStreamOpener, string id = null)
```

Comment:
I’m changing all of these to make the Id a required param. The configs will needs to be changed to include the ids, as well as the code that sets up the operations.

From: Mike Lorbetske [mailto:notifications@github.com]
Sent: Tuesday, August 23, 2016 11:17 AM
To: seancpeters/microsoft-templateengine <microsoft-templateengine@noreply.github.com>
Cc: Sean Peters <sepeters@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [seancpeters/microsoft-templateengine] Refactor (#1)


In src/Microsoft.TemplateEngine.Core/Include.cs<https://github.com/seancpeters/microsoft-templateengine/pull/1#discussion_r75921868>:

> @@ -8,11 +8,12 @@ namespace Microsoft.TemplateEngine.Core

>  {

>      public class Include : IOperationProvider

>      {

> -        public Include(string startToken, string endToken, Func<string, Stream> sourceStreamOpener)

> +        public Include(string startToken, string endToken, Func<string, Stream> sourceStreamOpener, string id = null)

Since these are all constructed from a single codepath, defaults and/or overloads aren't preferred

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/seancpeters/microsoft-templateengine/pull/1/files/a425b2f9e55c8aa446463bd90060cb6bc99db3d0#r75921868>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AT8a8aq1yq7x42hZ3q9Skm8RPvKd4juhks5qizkBgaJpZM4JqaeQ>.